stratosphere/loader/source/ldr_process_creation.cpp

```diff
@@ -14,248 +14,698 @@
+                        aslr_size  = map::AslrSize64BitDeprecated;
+                        break;
+                    case svc::CreateProcessFlag_AddressSpace64Bit:
+                        aslr_start = map::AslrBase64Bit;
+                        aslr_size  = map::AslrSize64Bit;
+                        break;
+                    default:
+                        std::abort();
+                }
+            } else {
+                /* On 1.0.0, only 2 address space types existed. */
+                if (out_cpi->flags & svc::CreateProcessFlag_AddressSpace64BitDeprecated) {
+                    aslr_start = map::AslrBase64BitDeprecated;
+                    aslr_size  = map::AslrSize64BitDeprecated;
+                } else {
+                    aslr_start = map::AslrBase32Bit;
+                    aslr_size  = map::AslrSize32Bit;
+                }
+            }
+            if (total_size > aslr_size) {
+                return ResultKernelOutOfMemory;
+            }
+
+            /* Set Create Process output. */
+            uintptr_t aslr_slide = 0;
+            uintptr_t unused_size = (aslr_size - total_size);
+            if (out_cpi->flags & svc::CreateProcessFlag_EnableAslr) {
+                aslr_slide = sts::rnd::GenerateRandomU64(unused_size / BaseAddressAlignment) * BaseAddressAlignment;
+            }
+
+            /* Set out. */
+            aslr_start += aslr_slide;
+            for (size_t i = 0; i < Nso_Count; i++) {
+                if (has_nso[i]) {
+                    out->nso_address[i] += aslr_start;
+                }
+            }
+            if (out->args_address) {
+                out->args_address += aslr_start;
+            }
+
+            out_cpi->code_address = aslr_start;
+            out_cpi->code_num_pages = total_size >> 12;
+
+            return ResultSuccess;
         }
-    };
 
-    /* Load the process's NPDM. */
-    R_TRY(NpdmUtils::LoadNpdmFromCache(target_process->tid_sid.title_id, &npdm_info));
+        Result CreateProcessImpl(ProcessInfo *out, const Meta *meta, const NsoHeader *nso_headers, const bool *has_nso, const args::ArgumentInfo *arg_info, u32 flags, Handle reslimit_h) {
+            /* Get CreateProcessInfo. */
+            CreateProcessInfo cpi;
+            R_TRY(GetCreateProcessInfo(&cpi, meta, flags, reslimit_h));
 
-    /* Validate version. */
-    R_TRY(ValidateProcessVersion(target_process->tid_sid.title_id, npdm_info.header->version));
+            /* Decide on an NSO layout. */
+            R_TRY(DecideAddressSpaceLayout(out, &cpi, nso_headers, has_nso, arg_info));
 
-    /* Validate the title we're loading is what we expect. */
-    if (npdm_info.aci0->title_id < npdm_info.acid->title_id_range_min || npdm_info.aci0->title_id > npdm_info.acid->title_id_range_max) {
-        return ResultLoaderInvalidProgramId;
-    }
+            /* Actually create process. const_cast necessary because libnx doesn't declare svcCreateProcess with const u32*. */
```

Comment:
Might be worth submitting a PR to libnx to fix this?

Reply:
Good call.